Modules/Devices/test/Devices.Application.Tests/Tests/Identities/Queries/GetIdentityByAddress/HandlerTests.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Backbone.Modules.Devices.Application.Identities.Queries.GetIdentityByAddress;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using Enmeshed.UnitTestTools.Extensions;
+using FakeItEasy;
+using FluentAssertions;
+using Xunit;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Identities.Queries.GetIdentityByAddress;
+public class HandlerTests
+{
+    [Fact]
+    public async void Gets_identity_by_address()
+    {
+        // Arrange
+        var expectedClientId = TestDataGenerator.CreateRandomDeviceId();
+        var expectedAddress = TestDataGenerator.CreateRandomIdentityAddress();
+        var expectedTierId = TestDataGenerator.CreateRandomTierId();
+        var identity = new Identity(expectedClientId, expectedAddress, Array.Empty<byte>(), expectedTierId, 1);
+
+        var handler = CreateHandler(new FindByAddressStubRepository(identity));
+
+        // Act
+        var result = await handler.Handle(new GetIdentityByAddressQuery(identity.Address), CancellationToken.None);
+
+        // Assert
+        result.ClientId.Should().Be(expectedClientId);
+        result.Address.Should().Be(expectedAddress);
+        result.PublicKey.Should().BeEquivalentTo(Array.Empty<byte>());
```

Comment:
Using an empty array is not ideal, as it might as well be empty because it is not mapped correctly.